glambie/processing/process_global_results.py

```diff
@@ -105,32 +109,42 @@ def _combine_regional_results_into_global(regional_results_catalogue: DataCatalo
     Returns
     -------
     Timeseries
-        Global timeseries
+        Globally aggregated timeseries
     """
-
-    # TODO: need to adapt error propagation in this
-
     assert regional_results_catalogue.datasets_are_same_unit()
     assert regional_results_catalogue.datasets[0].unit == "mwe"
 
-    # Multiply by area to calculate area weighted mean
-    regional_results_catalogue = regional_results_catalogue.copy()  # make sure we dont edit existing datasets
-    total_area = 0
-    for ds in regional_results_catalogue.datasets:
-        ds.data.changes = ds.data.changes * ds.region.rgi6_area
-        ds.data.errors = ds.data.changes * ds.region.rgi6_area
-        total_area = total_area + ds.region.rgi6_area
-
-    # combine all regions into global
-    combined_ts, _ = regional_results_catalogue.average_timeseries_in_catalogue(remove_trend=True,
-                                                                                add_trend_after_averaging=True,
-                                                                                out_data_group=GLAMBIE_DATA_GROUPS[
-                                                                                    "consensus"])
-    # divide by area
-    combined_ts.data.changes = combined_ts.data.changes / (total_area / len(regional_results_catalogue.datasets))
-    # PLACEHOLDER FOR ERROR CALC FOR NOW
-    combined_ts.data.errors = combined_ts.data.errors * 0
-
-    combined_ts.region = REGIONS["global"]
-
-    return combined_ts
+    # merge all dataframes
+    catalogue_dfs = [ds.data.as_dataframe() for ds in regional_results_catalogue.datasets]
+    # calculate global area for weighted mean
+    total_area = np.sum([ds.region.rgi6_area for ds in regional_results_catalogue.datasets])
+
+    # multiply changes and errors with area for each region
+    for _, (df, ds) in enumerate(zip(catalogue_dfs, regional_results_catalogue.datasets)):
+        df["changes"] = (df["changes"] * ds.region.rgi6_area)
+        # apply weighted mean error propagation
+        df["errors"] = (df["errors"] * ds.region.rgi6_area)**2
+        # ds.data.errors = ds.data.changes * ds.region.rgi6_area
+    # calculate weighted mean
+    df = reduce(lambda left, right: left.merge(right, how="outer", on=["start_dates", "end_dates"]), catalogue_dfs)
+    df = df.sort_values(by="start_dates")
+    start_dates, end_dates = np.array(df["start_dates"]), np.array(df["end_dates"])
+    mean_changes = np.array(df[df.columns.intersection(
+        df.filter(regex=("changes*")).columns.to_list())].sum(axis=1)) / total_area
+
+    # apply square root to errors and divide by total area
+    mean_uncertainties = np.sqrt(np.array(df[df.columns.intersection(
+        df.filter(regex=("errors*")).columns.to_list())].sum(axis=1))) / total_area
+
+    # make timeseries object with combined solution
+    ts_data = TimeseriesData(start_dates=np.array(start_dates),
```

Comment:
Already cast to np.array above?
```suggestion
    ts_data = TimeseriesData(start_dates=start_dates,
```

Reply:
yes! done